src/libdredd/src/mutate_visitor.cc

```diff
@@ -508,6 +524,7 @@ bool MutateVisitor::VisitExpr(clang::Expr* expr) {
     return true;
   }
 
+  UpdateStartLocationOfFirstFunctionInSourceFile();
```

Comment:
I think it would be cleaner to call this whenever a mutation is added.

I propose:

- introduce a private helper function, `AddMutation`
- have the body of `AddMutation` do the `mutation_tree_path_.back()->AddMutation(...)` call that is currently made a few times in `mutate_visitor.cc`, and replace those calls with a call to the new `AddMutation` method
- Have the new `AddMutation` method call `UpdateStartLocationOfFirstFunctionInSourceFile` as the first thing that it does.

I think this would be good because if in the future we add a new kind of mutation to Dredd, we won't need to remember to add calls to `UpdateStartLocation...`.